### PR TITLE
Updates the CAPV version to v1.0.2

### PR DESCRIPTION
**Pull request by srm09**
### What this PR does / why we need it
This patch has fixes for:
- Deletion of VSphereDeploymentZone & VSphereFailureDomain CRs
- Fixes naming association for VSphereDeploymentZone with
  spec.FailureDomain for Machine objects

### Which issue(s) this PR fixes
Fixes # n/a

### Describe testing done for PR
Upstream CI tests were run for the changes

### Release note
```release-note
Update CAPV to version v1.0.2
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### Additional information
n/a

#### Special notes for your reviewer
n/a

### CODE REVIEW

**Comment by srm09** [comment]
cc: @AartiJivrajani for the sake of visibility.

**Review by danniel1205 — APPROVED** [review]
LGTM. Adding approval to unblock it from merging, it would be great to get additional approvals from component owners as well.

**Review by vuil — APPROVED** [review]
lgtm, thanks.

**Pull request merged**
